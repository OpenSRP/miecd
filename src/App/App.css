tr.definitions td:first-child {
    padding-left: 0px;
}

td {
    font-size: 10;
}

h2.page-title {
    font-weight: 200;
}

.gatekeeper-cb,
.gatekeeper-login {
    margin-top: 3rem !important;
    padding: 2%;
}

.gatekeeper-btn {
    padding: 10px 16px;
    font-size: 18px;
    line-height: 1.33;
    border-radius: 6px;
    color: #ffffff;
    background-color: #28a745;
    border-color: #28a745;
    width: 100%;
    display: inline-block;
    max-width: 200px;
}

.gatekeeper-btn:hover,
.gatekeeper-btn:focus,
.gatekeeper-btn:active,
.gatekeeper-btn.active,
.open .dropdown-toggle.gatekeeper-btn {
    color: #ffffff;
    background-color: #169c34;
    border-color: #28a745;
    text-decoration: none;
}

.gatekeeper-btn:active,
.gatekeeper-btn.active,
.open .dropdown-toggle.gatekeeper-btn {
    background-image: none;
}

.gatekeeper-btn.disabled,
.gatekeeper-btn[disabled],
fieldset[disabled] .gatekeeper-btn,
.gatekeeper-btn.disabled:hover,
.gatekeeper-btn[disabled]:hover,
fieldset[disabled] .gatekeeper-btn:hover,
.gatekeeper-btn.disabled:focus,
.gatekeeper-btn[disabled]:focus,
fieldset[disabled] .gatekeeper-btn:focus,
.gatekeeper-btn.disabled:active,
.gatekeeper-btn[disabled]:active,
fieldset[disabled] .gatekeeper-btn:active,
.gatekeeper-btn.disabled.active,
.gatekeeper-btn[disabled].active,
fieldset[disabled] .gatekeeper-btn.active {
    background-color: #28a745;
    border-color: #28a745;
}

.gatekeeper-btn .badge {
    color: #28a745;
    background-color: #ffffff;
}

.compartment-wrapper.compartments {
    padding-left: 24px;
    padding-right: 48px;
}

.sidebar {
    width: 18.5vw;
}

.hidden-container {
    display: none !important;
}

.main-app-container {
    height: 100vh;
    border: 0;
    margin: 0;
    padding: 0;
    font-family: 'Roboto', 'Helvetica';
}

.main-app-container .main-container {
    min-height: 100vh;
}

.main-app-container .navbar {
    height: 60px;
    position: fixed;
    z-index: 1;
    width: 100%;
}

.main-app-container .content {
    width: 100%;
    position: relative;
    top: 60px;
}

.main-page-row {
    background-color: #f8f9fd;
}
