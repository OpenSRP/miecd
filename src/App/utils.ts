import { DEFAULT_PLAN_ID, KEYCLOAK_API_BASE_URL, KEYCLOAK_USERS_PAGE_SIZE, OPENSRP_API_BASE_URL } from 'configs/env';
import { URL_LOCATION_UNIT } from '../constants';

const openSrpRestBase = OPENSRP_API_BASE_URL + '/rest/';

export const baseProps = {
    baseURL: openSrpRestBase,
    opensrpBaseURL: openSrpRestBase,
    keycloakBaseURL: KEYCLOAK_API_BASE_URL,
};

export const locationUnitProps = {
    ...baseProps,
    filterByParentId: false,
};

export const newLocationUnitProps = {
    ...locationUnitProps,
    successURLGenerator: () => URL_LOCATION_UNIT,
    cancelURLGenerator: () => URL_LOCATION_UNIT,
    hidden: ['serviceType', 'latitude', 'longitude'],
};

export const editLocationProps = {
    ...newLocationUnitProps,
    ...locationUnitProps,
};

export const usersListProps = {
    ...baseProps,
    usersPageSize: KEYCLOAK_USERS_PAGE_SIZE,
};

export const teamAssignmentProps = {
    ...baseProps,
    defaultPlanId: DEFAULT_PLAN_ID,
};

export const createEditUserProps = {
    ...baseProps,
    userFormHidden: [],
};
