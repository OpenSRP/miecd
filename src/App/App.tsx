import { library } from '@fortawesome/fontawesome-svg-core';
import { faUser } from '@fortawesome/free-regular-svg-icons';
import { faArrowLeft, faChartLine, faCog, faHome } from '@fortawesome/free-solid-svg-icons';
import React from 'react';
import { Route } from 'react-router';
import { providers } from '../configs/settings';
import { LOGIN_URL } from '../constants';
import ConnectedHeader from '../containers/ConnectedHeader';
import { headerShouldRender } from '../helpers/utils';
import './App.css';
import CustomOauthLogin from '../components/CustomAuthLogin';
import ConnectedRoutes from './Routes';
import { I18nextProvider } from 'react-i18next';
import i18n from '../mls';
import { Toaster } from 'react-hot-toast';

library.add(faUser, faChartLine, faCog, faHome, faArrowLeft);

/** Main App component */
export const App = () => {
    return (
        <I18nextProvider i18n={i18n}>
            <div className="main-app-container">
                <Route
                    exact
                    path={LOGIN_URL}
                    // tslint:disable-next-line: jsx-no-lambda
                    render={(routeProps) => <CustomOauthLogin providers={providers} {...routeProps} />}
                />
                {/* tslint:enable jsx-no-lambda */}
                <ConnectedHeader />
                {headerShouldRender() && <ConnectedRoutes />}
                <Toaster />
            </div>
        </I18nextProvider>
    );
};

export default App;
