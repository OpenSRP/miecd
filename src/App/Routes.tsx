import React from 'react';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faUser } from '@fortawesome/free-regular-svg-icons';
import ConnectedPrivateRoute from '@onaio/connected-private-route';
import { ConnectedLogout, ConnectedOauthCallback, LogoutProps } from '@onaio/gatekeeper';
import { isAuthenticated } from '@onaio/session-reducer';
import { connect } from 'react-redux';
import { Store } from 'redux';
import { Route, RouteComponentProps, Switch } from 'react-router';
import { LastLocationProvider } from 'react-router-last-location';
import Loading from '../components/page/Loading';
import SideMenu from '../components/page/SideMenu';
import {
    NBC_AND_PNC_ANALYSIS_ENDPOINT,
    OPENSRP_LOGOUT_URL,
    SUPERSET_PREGNANCY_ANALYSIS_ENDPOINT,
} from '../configs/env';
import { providers } from '../configs/settings';
import {
    CHILD_PATIENT_DETAIL_URL,
    HIERARCHICAL_DATA_URL,
    LOGOUT_URL,
    NBC_AND_PNC_ANALYSIS_URL,
    NBC_AND_PNC_COMPARTMENTS_URL,
    NBC_AND_PNC_LOGFACE_URL,
    NBC_AND_PNC_MODULE,
    NBC_AND_PNC_URL,
    NUTRITION_ANALYSIS_URL,
    NUTRITION_COMPARTMENTS_URL,
    NUTRITION_LOGFACE_URL,
    NUTRITION_MODULE,
    NUTRITION_URL,
    PATIENT_DETAIL_URL,
    PREGNANCY_ANALYSIS_URL,
    PREGNANCY_COMPARTMENTS_URL,
    PREGNANCY_LOGFACE_URL,
    PREGNANCY_MODULE,
    PREGNANCY_URL,
    PREGNANCY,
    NUTRITION,
    NBC_AND_PNC,
} from '../constants';
import Compartments from '../containers/Compartments';
import ConnectedHierarchicalDataTable from '../containers/HierarchichalDataTable';
import ConnectedLogFace from '../containers/LogFace';
import Analysis from '../containers/pages/Analysis';
import Home from '../containers/pages/Home';
import ModuleHome from '../containers/pages/ModuleHome';
import ConnectedPatientDetails from '../containers/PatientDetails';
import { headerShouldRender, oAuthUserInfoGetter } from '../helpers/utils';
import './App.css';
import { Trans, useTranslation } from 'react-i18next';

library.add(faUser);

const mapStateToProps = (state: Partial<Store>) => {
    return {
        authenticated: isAuthenticated(state),
    };
};

export interface RoutesProps {
    authenticated: boolean;
}

export const Routes = (props: RoutesProps) => {
    const { authenticated } = props;
    const { t } = useTranslation();
    const NBC_AND_PNC_DASHBOARD_WELCOME = t('Welcome to Newborn and Postnatal Care');
    const NUTRITION_DASHBOARD_WELCOME = t('Welcome to Nutrition Care');
    const PREGNANCY_DASHBOARD_WELCOME = t('Welcome to the pregnancy dashboard');
    const PREGNANCY_DESCRIPTION = (
        <Trans>
            This dashboard displays information collected from MIECD Viet Nam Pregnancy Module for patients in your
            geographical location.&nbsp;
            <span className="font-italic">
                The module covers the whole pregnancy period from conception to delivery and includes Pregnancy
                Registration, ANC visits, Birth/Death reports, Risk Reports, Risk alerts and Response reports.
            </span>
        </Trans>
    );

    return (
        <div className={`${authenticated && headerShouldRender() ? 'main-container' : 'hidden-container'}`}>
            <SideMenu authenticated={authenticated} />
            <div className="content">
                <Switch>
                    <LastLocationProvider>
                        <ConnectedPrivateRoute disableLoginProtection={false} exact path="/" component={Home} />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={PREGNANCY_URL}
                            // tslint:disable-next-line: jsx-no-lambda
                            component={() => (
                                <ModuleHome
                                    title={PREGNANCY_DASHBOARD_WELCOME}
                                    description={PREGNANCY_DESCRIPTION}
                                    logFaceUrl={PREGNANCY_LOGFACE_URL}
                                    compartmentUrl={PREGNANCY_COMPARTMENTS_URL}
                                    analysisUrl={PREGNANCY_ANALYSIS_URL}
                                />
                            )}
                        />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={NBC_AND_PNC_URL}
                            // tslint:disable-next-line: jsx-no-lambda
                            component={() => (
                                <ModuleHome
                                    title={NBC_AND_PNC_DASHBOARD_WELCOME}
                                    description={PREGNANCY_DESCRIPTION}
                                    logFaceUrl={NBC_AND_PNC_LOGFACE_URL}
                                    compartmentUrl={NBC_AND_PNC_COMPARTMENTS_URL}
                                    analysisUrl={NBC_AND_PNC_ANALYSIS_URL}
                                />
                            )}
                        />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={NUTRITION_URL}
                            // tslint:disable-next-line: jsx-no-lambda
                            component={() => (
                                <ModuleHome
                                    title={NUTRITION_DASHBOARD_WELCOME}
                                    description={PREGNANCY_DESCRIPTION}
                                    logFaceUrl={NUTRITION_LOGFACE_URL}
                                    compartmentUrl={NUTRITION_COMPARTMENTS_URL}
                                    analysisUrl={NUTRITION_ANALYSIS_URL}
                                />
                            )}
                        />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={PREGNANCY_COMPARTMENTS_URL}
                            // tslint:disable-next-line: jsx-no-lambda
                            component={() => <Compartments module={PREGNANCY_MODULE} />}
                        />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={NBC_AND_PNC_COMPARTMENTS_URL}
                            // tslint:disable-next-line: jsx-no-lambda
                            component={() => <Compartments module={NBC_AND_PNC_MODULE} />}
                        />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={NUTRITION_COMPARTMENTS_URL}
                            // tslint:disable-next-line: jsx-no-lambda
                            component={() => <Compartments module={NUTRITION_MODULE} />}
                        />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={(() => {
                                return [
                                    NUTRITION_COMPARTMENTS_URL,
                                    NBC_AND_PNC_COMPARTMENTS_URL,
                                    PREGNANCY_COMPARTMENTS_URL,
                                ].map(
                                    (url) =>
                                        `${url}${HIERARCHICAL_DATA_URL}/:module?/:risk_highlighter?/:title?/:current_level?/:direction?/:node_id?/:permission_level?/:from_level?`,
                                );
                            })()}
                            component={ConnectedHierarchicalDataTable}
                        />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={PREGNANCY_ANALYSIS_URL}
                            // tslint:disable-next-line: jsx-no-lambda
                            component={() => (
                                <Analysis endpoint={SUPERSET_PREGNANCY_ANALYSIS_ENDPOINT} module={PREGNANCY} />
                            )}
                        />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={NBC_AND_PNC_ANALYSIS_URL}
                            // tslint:disable-next-line: jsx-no-lambda
                            component={() => <Analysis endpoint={NBC_AND_PNC_ANALYSIS_ENDPOINT} module={NBC_AND_PNC} />}
                        />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={NUTRITION_ANALYSIS_URL}
                            // tslint:disable-next-line: jsx-no-lambda
                            component={() => (
                                <Analysis endpoint={SUPERSET_PREGNANCY_ANALYSIS_ENDPOINT} module={NUTRITION} />
                            )}
                        />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={(() => {
                                return [
                                    NUTRITION_COMPARTMENTS_URL,
                                    NBC_AND_PNC_COMPARTMENTS_URL,
                                    PREGNANCY_COMPARTMENTS_URL,
                                ].map((url) => `${url}${PATIENT_DETAIL_URL}`);
                            })()}
                            component={ConnectedPatientDetails}
                        />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={(() => {
                                return [
                                    NUTRITION_COMPARTMENTS_URL,
                                    NBC_AND_PNC_COMPARTMENTS_URL,
                                    PREGNANCY_COMPARTMENTS_URL,
                                ].map((url) => `${url}${CHILD_PATIENT_DETAIL_URL}`);
                            })()}
                            // tslint:disable-next-line: jsx-no-lambda
                            // eslint-disable-next-line @typescript-eslint/no-explicit-any
                            component={(routeProps: any) => <ConnectedPatientDetails isChild {...routeProps} />}
                        />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={(() => {
                                return [NUTRITION_LOGFACE_URL, NBC_AND_PNC_LOGFACE_URL, PREGNANCY_LOGFACE_URL].map(
                                    (url) => `${url}${PATIENT_DETAIL_URL}`,
                                );
                            })()}
                            component={ConnectedPatientDetails}
                        />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={(() => {
                                return [NUTRITION_LOGFACE_URL, NBC_AND_PNC_LOGFACE_URL, PREGNANCY_LOGFACE_URL].map(
                                    (url) => `${url}${CHILD_PATIENT_DETAIL_URL}`,
                                );
                            })()}
                            // tslint:disable-next-line: jsx-no-lambda
                            // eslint-disable-next-line @typescript-eslint/no-explicit-any
                            component={(routeProps: any) => <ConnectedPatientDetails isChild {...routeProps} />}
                        />
                        <ConnectedPrivateRoute
                            disableLoginProtection={false}
                            exact
                            path={LOGOUT_URL}
                            // tslint:disable-next-line: jsx-no-lambda
                            component={() => (
                                <ConnectedLogout {...({ logoutURL: OPENSRP_LOGOUT_URL } as Partial<LogoutProps>)} />
                            )}
                        />

                        <ConnectedPrivateRoute
                            exact
                            path={PREGNANCY_LOGFACE_URL}
                            // tslint:disable-next-line: jsx-no-lambda
                            component={(routeProps: RouteComponentProps) => (
                                <ConnectedLogFace module={PREGNANCY_MODULE} {...routeProps} />
                            )}
                        />
                        <ConnectedPrivateRoute
                            exact
                            path={NBC_AND_PNC_LOGFACE_URL}
                            // tslint:disable-next-line: jsx-no-lambda
                            component={(routeProps: RouteComponentProps) => (
                                <ConnectedLogFace module={NBC_AND_PNC_MODULE} {...routeProps} />
                            )}
                        />
                        <ConnectedPrivateRoute
                            exact
                            path={NUTRITION_LOGFACE_URL}
                            // tslint:disable-next-line: jsx-no-lambda
                            component={(routeProps: RouteComponentProps) => (
                                <ConnectedLogFace module={NUTRITION_MODULE} {...routeProps} />
                            )}
                        />
                        {/* tslint:disable jsx-no-lambda */}
                        <Route
                            exact
                            path="/oauth/callback/:id"
                            render={(routeProps) => (
                                <ConnectedOauthCallback
                                    LoadingComponent={Loading}
                                    providers={providers}
                                    oAuthUserInfoGetter={oAuthUserInfoGetter}
                                    SuccessfulLoginComponent={Home}
                                    {...routeProps}
                                />
                            )}
                        />
                        {/* tslint:enable jsx-no-lambda */}
                    </LastLocationProvider>
                </Switch>
            </div>
        </div>
    );
};

const ConnectedRoutes = connect(mapStateToProps)(Routes);

export default ConnectedRoutes;
