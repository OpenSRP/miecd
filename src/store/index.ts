import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';
import { Dictionary } from '@onaio/utils';

/** Initial reducers in the reducer registry */
const defaultReducers: Dictionary = {
    router: connectReducer /** Dirty hack because connectRouter LocationChangeAction does not extend Redux.AnyAction */,
};

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;

/** The initial store for the reveal web app */
const store = getConnectedStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener((reducers) => {
    store.replaceReducer(combine(reducers));
});

export default store;
