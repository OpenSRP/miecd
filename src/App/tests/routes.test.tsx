import { authenticateUser } from '@onaio/session-reducer';
import {
    PREGNANCY_ANALYSIS_URL,
    NBC_AND_PNC_ANALYSIS_URL,
    NUTRITION_ANALYSIS_URL,
    URL_TEAMS,
    URL_TEAMS_ADD,
    URL_TEAMS_EDIT,
    URL_TEAM_ASSIGNMENT,
    URL_LOCATION_UNIT,
    URL_LOCATION_UNIT_ADD,
    URL_LOCATION_UNIT_GROUP,
    URL_LOCATION_UNIT_GROUP_ADD,
} from '../../constants';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { MemoryRouter, RouteComponentProps } from 'react-router';
import store from '../../store';
import App from '../App';
import toJson from 'enzyme-to-json';
import React from 'react';
import {
    URL_USER_CREDENTIALS,
    URL_USER_EDIT,
    URL_USER_GROUP_CREATE,
    URL_USER_GROUP_EDIT,
    URL_USER,
    URL_USER_GROUPS,
} from '@opensrp/user-management';

jest.mock('containers/pages/Analysis', () => {
    const Analysis = () => {
        return <div>This is Analysis</div>;
    };
    return Analysis;
});

jest.mock('@opensrp/user-management', () => {
    const userMocks = jest.requireActual('@opensrp/user-management');
    const MockUsersComponent = ({ message }: { message: string }) => <div>{message}</div>;
    return {
        ...userMocks,
        ConnectedUserList: () => <MockUsersComponent message="Users list" />,
        ConnectedCreateEditUser: () => <MockUsersComponent message="UserEdit" />,
        ConnectedUserCredentials: () => <MockUsersComponent message="User credentials list" />,
        UserGroupsList: () => <MockUsersComponent message="User groups list" />,
        UserRolesList: () => <MockUsersComponent message="User roles list" />,
        CreateEditUserGroup: () => <MockUsersComponent message="Edit user groups" />,
    };
});

jest.mock('@opensrp/team-management', () => {
    const userMocks = jest.requireActual('@opensrp/team-management');
    const MockTeamsComponent = ({ message }: { message: string }) => <div>{message}</div>;
    return {
        ...userMocks,
        TeamsView: () => <MockTeamsComponent message="Teams list" />,
        TeamsAddEdit: () => <MockTeamsComponent message="Teams add/edit" />,
    };
});

jest.mock('@opensrp/team-assignment', () => {
    const userMocks = jest.requireActual('@opensrp/team-assignment');
    const MockTeamAssignmentComponent = ({ message }: { message: string }) => <div>{message}</div>;
    return {
        ...userMocks,
        TeamAssignmentView: () => <MockTeamAssignmentComponent message="Teams assignment" />,
    };
});

jest.mock('@opensrp/location-management', () => {
    const userMocks = jest.requireActual('@opensrp/location-management');
    const MockLocationComponent = ({ message }: { message: string }) => <div>{message}</div>;
    return {
        ...userMocks,
        LocationUnitList: () => <MockLocationComponent message="Location list" />,
        LocationUnitGroupAddEdit: () => <MockLocationComponent message="Location group add edit" />,
        LocationUnitGroupList: () => <MockLocationComponent message="Location unit group list" />,
        NewLocationUnit: () => <MockLocationComponent message="New location unit" />,
        EditLocationUnit: () => <MockLocationComponent message="Edit location unit" />,
    };
});

jest.mock('../../configs/env');

describe('App', () => {
    beforeAll(() => {
        store.dispatch(
            authenticateUser(
                true,
                {
                    email: 'test@gmail.com',
                    name: 'This Name',
                    username: 'tHat Part',
                },
                {
                    roles: ['ROLE_VIEW_KEYCLOAK_USERS'],
                    username: 'superset-user',
                    user_id: 'cab07278-c77b-4bc7-b154-bcbf01b7d35b',
                },
            ),
        );
    });

    beforeEach(() => {
        jest.resetAllMocks();
    });

    it('Analysis routes are correctly registered', async () => {
        // redirecting to certain routes renders the correct page
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={[{ pathname: PREGNANCY_ANALYSIS_URL }]}>
                    <App />
                </MemoryRouter>
            </Provider>,
        );
        await act(async () => {
            await new Promise<unknown>((resolve) => setImmediate(resolve));
            wrapper.update();
        });

        expect(toJson(wrapper.find('Analysis'))).toMatchSnapshot('pregnancy module');

        // go to new nbc and pnc module page
        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(NBC_AND_PNC_ANALYSIS_URL);
        wrapper.update();
        expect(toJson(wrapper.find('Analysis'))).toMatchSnapshot('nbc and pnc module');

        // go to edit nutrition module
        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(NUTRITION_ANALYSIS_URL);
        wrapper.update();
        expect(toJson(wrapper.find('Analysis'))).toMatchSnapshot('nutrition module');
        wrapper.unmount();
    });

    it('userManagement routes are correctly registered', async () => {
        // redirecting to certain routes renders the correct page
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={[{ pathname: URL_USER }]}>
                    <App />
                </MemoryRouter>
            </Provider>,
        );
        await act(async () => {
            await new Promise<unknown>((resolve) => setImmediate(resolve));
            wrapper.update();
        });

        expect(toJson(wrapper.find('MockUsersComponent'))).toMatchSnapshot('user list');

        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(`${URL_USER_EDIT}/uuid`);
        wrapper.update();
        expect(toJson(wrapper.find('MockUsersComponent'))).toMatchSnapshot();

        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(`${URL_USER_CREDENTIALS}/uuid`);
        wrapper.update();
        expect(toJson(wrapper.find('MockUsersComponent'))).toMatchSnapshot();

        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(`${URL_USER_GROUP_EDIT}/uuid`);
        wrapper.update();
        expect(toJson(wrapper.find('MockUsersComponent'))).toMatchSnapshot();

        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(`${URL_USER_GROUPS}/uuid`);
        wrapper.update();
        expect(toJson(wrapper.find('MockUsersComponent'))).toMatchSnapshot();

        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(URL_USER_GROUP_CREATE);
        wrapper.update();
        expect(toJson(wrapper.find('MockUsersComponent'))).toMatchSnapshot();
        wrapper.unmount();
    });

    it('teams routes are correctly registered', async () => {
        // redirecting to certain routes renders the correct page
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={[{ pathname: URL_TEAMS }]}>
                    <App />
                </MemoryRouter>
            </Provider>,
        );
        await act(async () => {
            await new Promise<unknown>((resolve) => setImmediate(resolve));
            wrapper.update();
        });

        expect(toJson(wrapper.find('MockTeamsComponent'))).toMatchSnapshot('user list');

        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(URL_TEAM_ASSIGNMENT);
        wrapper.update();
        expect(toJson(wrapper.find('MockTeamAssignmentComponent'))).toMatchSnapshot();

        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(`${URL_TEAMS_EDIT}/uuid`);
        wrapper.update();
        expect(toJson(wrapper.find('MockTeamsComponent'))).toMatchSnapshot();

        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(URL_TEAMS_ADD);
        wrapper.update();
        expect(toJson(wrapper.find('MockTeamsComponent'))).toMatchSnapshot();
        wrapper.unmount();
    });

    it('location units routes are correctly registered', async () => {
        // redirecting to certain routes renders the correct page
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={[{ pathname: URL_LOCATION_UNIT }]}>
                    <App />
                </MemoryRouter>
            </Provider>,
        );
        await act(async () => {
            await new Promise<unknown>((resolve) => setImmediate(resolve));
            wrapper.update();
        });

        expect(toJson(wrapper.find('MockLocationComponent'))).toMatchSnapshot('user list');

        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(URL_LOCATION_UNIT_GROUP);
        wrapper.update();
        expect(toJson(wrapper.find('MockLocationComponent'))).toMatchSnapshot();

        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(
            `${URL_LOCATION_UNIT_GROUP}/edit/:id`,
        );
        wrapper.update();
        expect(toJson(wrapper.find('MockLocationComponent'))).toMatchSnapshot();

        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(URL_LOCATION_UNIT_GROUP_ADD);
        wrapper.update();
        expect(toJson(wrapper.find('MockLocationComponent'))).toMatchSnapshot();

        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(
            `${URL_LOCATION_UNIT}/edit/:id`,
        );
        wrapper.update();
        expect(toJson(wrapper.find('MockLocationComponent'))).toMatchSnapshot();

        (wrapper.find('Router').prop('history') as RouteComponentProps['history']).push(URL_LOCATION_UNIT_ADD);
        wrapper.update();
        expect(toJson(wrapper.find('MockLocationComponent'))).toMatchSnapshot();
        wrapper.unmount();
    });
});
