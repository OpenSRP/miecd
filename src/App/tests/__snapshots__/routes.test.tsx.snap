// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Analysis routes are correctly registered: nbc and pnc module 1`] = `
<Analysis
  module="NBC & PNC"
>
  <div>
    This is Analysis
  </div>
</Analysis>
`;

exports[`App Analysis routes are correctly registered: nutrition module 1`] = `
<Analysis
  module="Nutrition"
>
  <div>
    This is Analysis
  </div>
</Analysis>
`;

exports[`App Analysis routes are correctly registered: pregnancy module 1`] = `
<Analysis
  module="Pregnancy"
>
  <div>
    This is Analysis
  </div>
</Analysis>
`;

exports[`App location units routes are correctly registered 1`] = `
<MockLocationComponent
  message="Location unit group list"
>
  <div>
    Location unit group list
  </div>
</MockLocationComponent>
`;

exports[`App location units routes are correctly registered 2`] = `
<MockLocationComponent
  message="Location group add edit"
>
  <div>
    Location group add edit
  </div>
</MockLocationComponent>
`;

exports[`App location units routes are correctly registered 3`] = `
<MockLocationComponent
  message="Location group add edit"
>
  <div>
    Location group add edit
  </div>
</MockLocationComponent>
`;

exports[`App location units routes are correctly registered 4`] = `
<MockLocationComponent
  message="Edit location unit"
>
  <div>
    Edit location unit
  </div>
</MockLocationComponent>
`;

exports[`App location units routes are correctly registered 5`] = `
<MockLocationComponent
  message="New location unit"
>
  <div>
    New location unit
  </div>
</MockLocationComponent>
`;

exports[`App location units routes are correctly registered: user list 1`] = `
<MockLocationComponent
  message="Location list"
>
  <div>
    Location list
  </div>
</MockLocationComponent>
`;

exports[`App teams routes are correctly registered 1`] = `
<MockTeamAssignmentComponent
  message="Teams assignment"
>
  <div>
    Teams assignment
  </div>
</MockTeamAssignmentComponent>
`;

exports[`App teams routes are correctly registered 2`] = `
<MockTeamsComponent
  message="Teams add/edit"
>
  <div>
    Teams add/edit
  </div>
</MockTeamsComponent>
`;

exports[`App teams routes are correctly registered 3`] = `
<MockTeamsComponent
  message="Teams add/edit"
>
  <div>
    Teams add/edit
  </div>
</MockTeamsComponent>
`;

exports[`App teams routes are correctly registered: user list 1`] = `
<MockTeamsComponent
  message="Teams list"
>
  <div>
    Teams list
  </div>
</MockTeamsComponent>
`;

exports[`App userManagement routes are correctly registered 1`] = `
<MockUsersComponent
  message="UserEdit"
>
  <div>
    UserEdit
  </div>
</MockUsersComponent>
`;

exports[`App userManagement routes are correctly registered 2`] = `
<MockUsersComponent
  message="User credentials list"
>
  <div>
    User credentials list
  </div>
</MockUsersComponent>
`;

exports[`App userManagement routes are correctly registered 3`] = `
<MockUsersComponent
  message="Edit user groups"
>
  <div>
    Edit user groups
  </div>
</MockUsersComponent>
`;

exports[`App userManagement routes are correctly registered 4`] = `
<MockUsersComponent
  message="User groups list"
>
  <div>
    User groups list
  </div>
</MockUsersComponent>
`;

exports[`App userManagement routes are correctly registered 5`] = `
<MockUsersComponent
  message="Edit user groups"
>
  <div>
    Edit user groups
  </div>
</MockUsersComponent>
`;

exports[`App userManagement routes are correctly registered: user list 1`] = `
<MockUsersComponent
  message="Users list"
>
  <div>
    Users list
  </div>
</MockUsersComponent>
`;
